common/templates/context_funcs.go

```diff
@@ -868,3 +868,32 @@ func (c *Context) tmplOnlineCountBots() (int, error) {
 	return botCount, nil
 }
 
+func (c *Context) tmplEditNickname(Nickname string) (string, error) {
+
+	if c.IncreaseCheckCallCounter("edit_nick", 2) {
+		return "", ErrTooManyCalls
+	}
+
+		if c.MS == nil {
+		return "", nil
+	}
+
+	c.GS.RLock()
```

Comment:
No need to lock here, guild id is immutable and c.MS is a copy of the member when the cc started, not a reference.

Reply:
Made the changes. I was using sendDM as reference for some parts and it had the lock hence i thought it might be necessary.

Reply:
Yeah some pieces are old, used to be like that before but yeah